### ISSUE TITLE
--lib not being respected with `cargo generate`

### ISSUE THREAD

**Issue by midnightexigent** [rust-github/template#12]
Hi, 

Is it normal that I am getting `src/main.rs` even with `cargo generate --lib rust-github/template` ?

**Comment by marcoieni** [comment]
Yes, but with the latest release of cargo generate this can be solved. We have to include the file conditionally. 
If you want to contribute I can explain better. Otherwise I can have this done by next week.

**Comment by midnightexigent** [comment]
I can give it a shot

**Comment by marcoieni** [comment]
Oh, actually the feature was merged 10 days ago [here](https://github.com/cargo-generate/cargo-generate/commit/e91bfc9b2c7f30743273dde0ac5985b1bcfd9a44) but it is not released yet. I guess it will be released in the next version.

If you want starting working on it you can build `cargo generate` from the master branch. Otherwise we can wait until it is released :)

# How to solve the issue

## Ignore files conditionally

[Here](https://github.com/cargo-generate/cargo-generate#conditional-template-settings) there is the documentation on how to ignore files based on a condition.
I guess that in the file [cargo-generate.toml](https://github.com/rust-github/template/blob/main/cargo-generate.toml) we have to add something like:

```toml
[conditional.'crate_type == "lib"']
ignore = [ "src/main.rs" ]

[conditional.'crate_type == "bin"']
ignore = [ "src/lib.rs" ]
```

## Bump minimum version

This will be available since cargo-generate 0.10.0 ( or 0.9.1? 🤔), so you have to bump the version [here](https://github.com/rust-github/template/blob/c52ff451e80fcd626067d06f2d8d4f86f07b66d7/cargo-generate.toml#L2).

## Tip

For testing this locally you can use [this](https://github.com/cargo-generate/cargo-generate#generating-using-a-local-template).

**Comment by marcoieni** [comment]
closed by https://github.com/rust-github/template/pull/13
